src/useGet.tsx

```diff
@@ -0,0 +1,182 @@
+      }
+    | boolean
+    | number;
+}
+
+async function _fetchData<TData, TError, TQueryParams>(
+  props: UseGetProps<TData, TQueryParams>,
+  state: GetState<TData, TError>,
+  setState: (newState: GetState<TData, TError>) => void,
+  context: RestfulReactProviderProps,
+  abortControllers: AbortController[],
+) {
+  const { base = context.base, path, resolve = (d: any) => d as TData, queryParams } = props;
+
+  if (state.loading) {
+    // Abort previous requests
+    abortControllers.map(i => i.abort());
+
+    // Create a new abort controller
+    abortControllers.push(new AbortController());
+  }
+  const signal = abortControllers[abortControllers.length - 1].signal;
+
+  if (state.error || !state.loading) {
+    setState({ ...state, error: null, loading: true });
+  }
+
+  const requestOptions =
+    (typeof props.requestOptions === "function" ? props.requestOptions() : props.requestOptions) || {};
+  requestOptions.headers = new Headers(requestOptions.headers);
+
+  const request = new Request(
+    url.resolve(base, queryParams ? `${path}?${qs.stringify(queryParams)}` : path),
+    requestOptions,
+  );
+  try {
+    const response = await fetch(request, { signal });
+    const { data, responseError } = await processResponse(response);
+
+    if (signal.aborted) {
+      return;
+    } else if (!response.ok || responseError) {
+      const error = {
+        message: `Failed to fetch: ${response.status} ${response.statusText}${responseError ? " - " + data : ""}`,
+        data,
+        status: response.status,
+      };
+
+      setState({ ...state, loading: false, error });
+
+      if (!props.localErrorOnly && context.onError) {
+        context.onError(error, () => _fetchData(props, state, setState, context, abortControllers), response);
+      }
+    } else {
+      setState({ ...state, loading: false, data: resolve(data) });
+    }
+  } catch (e) {
+    setState({
+      ...state,
+      loading: false,
+      error: {
+        message: `Failed to fetch: ${e.message}`,
+        data: e.message,
+      },
+    });
+  }
+}
+
+export function useGet<TData = any, TError = any, TQueryParams = { [key: string]: any }>(
+  props: UseGetProps<TData, TQueryParams>,
+) {
+  const context = useContext(Context);
+
+  const fetchData = useCallback<typeof _fetchData>(
+    typeof props.debounce === "object"
+      ? debounce<typeof _fetchData>(_fetchData, props.debounce.wait, props.debounce.options)
+      : typeof props.debounce === "number"
+      ? debounce<typeof _fetchData>(_fetchData, props.debounce)
+      : props.debounce
+      ? debounce<typeof _fetchData>(_fetchData)
+      : _fetchData,
+    [props.debounce],
+  );
+
+  const [state, setState] = useState<GetState<TData, TError>>({
+    data: null,
+    response: null,
+    loading: !props.lazy,
+    error: null,
+  });
+
+  const abortControllers = useRef([new AbortController()]);
+
+  const prevProps = useRef<UseGetProps<TData, TQueryParams>>();
+  useEffect(() => {
```

Comment:
Please add a new line here to separate this from `prevProps`. I thought this was still the implementation of `prevProps` and had to scroll up/look twice.

```suggestion

  useEffect(() => {
```